job_test.go

```diff
@@ -278,3 +279,279 @@ func TestFilterJobs(t *testing.T) {
+		  <resource name="min_cpu_interval" type="qf">00:05:00</resource>
+		  <job_list state="running">
+			<JB_job_number>1005</JB_job_number>
+			<JAT_prio>0.55500</JAT_prio>
+			<JB_name>task_array.sh</JB_name>
+			<JB_owner>darrellb</JB_owner>
+			<state>r</state>
+			<JAT_start_time>2019-11-15T11:31:47</JAT_start_time>
+			<slots>1</slots>
+			<tasks>cat</tasks>
+		  </job_list>
+		  <job_list state="running">
+			<JB_job_number>1006</JB_job_number>
+			<JAT_prio>0.55500</JAT_prio>
+			<JB_name>task_array.sh</JB_name>
+			<JB_owner>darrellb</JB_owner>
+			<state>r</state>
+			<JAT_start_time>2019-11-15T11:31:47</JAT_start_time>
+			<slots>1</slots>
+			<tasks>10</tasks>
+		  </job_list>
+		  <job_list state="running">
+			<JB_job_number>1006</JB_job_number>
+			<JAT_prio>0.55500</JAT_prio>
+			<JB_name>task_array.sh</JB_name>
+			<JB_owner>darrellb</JB_owner>
+			<state>r</state>
+			<JAT_start_time>2019-11-15T11:31:47</JAT_start_time>
+			<slots>1</slots>
+			<tasks>15</tasks>
+		  </job_list>
+		  <job_list state="running">
+			<JB_job_number>1006</JB_job_number>
+			<JAT_prio>0.55500</JAT_prio>
+			<JB_name>task_array.sh</JB_name>
+			<JB_owner>darrellb</JB_owner>
+			<state>r</state>
+			<JAT_start_time>2019-11-15T11:31:47</JAT_start_time>
+			<slots>1</slots>
+			<tasks>20</tasks>
+		  </job_list>
+		  <job_list state="running">
+			<JB_job_number>1006</JB_job_number>
+			<JAT_prio>0.55500</JAT_prio>
+			<JB_name>task_array.sh</JB_name>
+			<JB_owner>darrellb</JB_owner>
+			<state>r</state>
+			<JAT_start_time>2019-11-15T11:31:47</JAT_start_time>
+			<slots>1</slots>
+			<tasks>25</tasks>
+		  </job_list>
+		  <job_list state="running">
+			<JB_job_number>1006</JB_job_number>
+			<JAT_prio>0.55500</JAT_prio>
+			<JB_name>task_array.sh</JB_name>
+			<JB_owner>darrellb</JB_owner>
+			<state>r</state>
+			<JAT_start_time>2019-11-15T11:31:47</JAT_start_time>
+			<slots>1</slots>
+			<tasks>30</tasks>
+		  </job_list>
+		  <job_list state="running">
+			<JB_job_number>1006</JB_job_number>
+			<JAT_prio>0.55500</JAT_prio>
+			<JB_name>task_array.sh</JB_name>
+			<JB_owner>darrellb</JB_owner>
+			<state>r</state>
+			<JAT_start_time>2019-11-15T11:31:47</JAT_start_time>
+			<slots>1</slots>
+			<tasks>35</tasks>
+		  </job_list>
+		  <job_list state="running">
+			<JB_job_number>1006</JB_job_number>
+			<JAT_prio>0.55500</JAT_prio>
+			<JB_name>task_array.sh</JB_name>
+			<JB_owner>darrellb</JB_owner>
+			<state>r</state>
+			<JAT_start_time>2019-11-15T11:31:47</JAT_start_time>
+			<slots>1</slots>
+			<tasks>40</tasks>
+		  </job_list>
+		</Queue-List>
+	  </queue_info>
+	  <job_info>
+		<job_list state="pending">
+		  <JB_job_number>1006</JB_job_number>
+		  <JAT_prio>0.55500</JAT_prio>
+		  <JB_name>task_array.sh</JB_name>
+		  <JB_owner>darrellb</JB_owner>
+		  <state>qw</state>
+		  <JB_submission_time>2019-11-15T11:31:34</JB_submission_time>
+		  <slots>1</slots>
+		  <tasks>41-150:1</tasks>
+		</job_list>
+	  </job_info>
+	</job_info>`
+
```

Comment:
What is this error supposed to be?

Reply:
The error actually comes from the fact the string doesn't match the regex for the range expression. I should add another assertion to verify it's the expected error.

Reply:
I created a variable to store the error in, and call it in the error scenario. Allows me to test to make sure that we both get the error, and that the error contents are as expected.